infra/src/batch-monitor.ts

```diff
@@ -0,0 +1,29 @@
+import { Stack, StackProps } from 'aws-cdk-lib';
+import * as events from 'aws-cdk-lib/aws-events';
+import * as evtTargets from 'aws-cdk-lib/aws-events-targets';
+import * as lf from 'aws-cdk-lib/aws-lambda';
+import { Code } from 'aws-cdk-lib/aws-lambda';
+import { Construct } from 'constructs';
+import * as path from 'path';
+
+export class AwsBatchMonitor extends Stack {
+  public constructor(scope: Construct, id: string, props: StackProps) {
+    super(scope, id, props);
+
+    const rule = new events.Rule(this, 'BatchEventRule', {
+      eventPattern: {
+        source: ['aws.batch'],
+        detailType: ['Batch Job State Change'],
+        detail: { status: ['FAILED'] },
```

Comment:
It might be worth it to just track all status changes? its super cheap to trace them all ? and might give us interesting information sometime in the future?